niceman/distributions/docker.py

```diff
@@ -0,0 +1,169 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the niceman package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Support for Docker distribution(s)."""
+
+import attr
+import json
+import logging
+
+lgr = logging.getLogger('niceman.distributions.docker')
+
+from .base import Package
+from .base import Distribution
+from .base import DistributionTracer
+from .base import TypedList
+from .base import _register_with_representer
+from ..dochelpers import borrowdoc
+from ..support.exceptions import CommandError
+
+
+@attr.s(slots=True, frozen=True, cmp=False, hash=True)
+class DockerImage(Package):
+    """Docker image information"""
+    id = attr.ib()
+    # Optional
+    architecture = attr.ib(default=None)
+    operating_system = attr.ib(default=None)
+    docker_version = attr.ib(default=None)
+    repo_digests = attr.ib(default=None)
+    repo_tags = attr.ib(default=None)
+    created = attr.ib(default=None)
```

Comment:
we will need an information about the hub from which it came (if possible)

Reply:
The hub (or in Docker speak, the registry and repository) where the image can be found is contained in the repo_digests property. For example, here is the digest for a local docker registry I am running on my local machine:

localhost:5000/my-ubuntu@sha256:615701b8ab7d454db039f412e50f22daa2af04b1561e08672e758cdac2af22cb

If there is no HTTP prefix to the digest, it is (by convention) defaulted to hub.docker.com

The hash after the @ symbol is a hash on the image manifest file located in the registry. The manifest contains, among other things, the name and tag of the image and the hashes of the file system layers needed to reconstruct the image.